### DISCUSSION TITLE
Filecoin on NodeJS

### DISCUSSION THREAD

**Discussion by ghost** [filecoin-project/community#513] [Developers]
Hello,
is there any way to use Filecoin with NodeJS?

The Quickstart guide that tells you to use Infura API is the opposite of what Filecoin should be, a decentralized storage for everyone on this planet. So I truly don't understand why they give you only one way "Infura API" to connect to the network.

I truly don't understand why the only NodeJS guide is a "bad" free tier 100k req/day API provider instead of having a connection to a gateway or to the blockchain or whatever.

**Comment by jilt** [comment]
I just published a template to use web3.storage on node.js with the near blockchain,
Beware, I had to modify some imports on the main npm package!
Find more info on the readme:
[Demo](https://jilt.github.io/filecoin-near-upload-app/)
[Repo](https://github.com/jilt/filecoin-near-upload-app)
